### PR TITLE
Update for Swift 4.1 and Xcode 9.3

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

File: Sources/Mapper.swift [hunk 1]
```diff
@@ -110,7 +110,7 @@ public struct Mapper {
         }
 
         let rawValues = try array.map { try T.RawValue.fromMap($0) }
-        return rawValues.flatMap { T(rawValue: $0) ?? defaultValue }
+        return rawValues.compactMap { T(rawValue: $0) ?? defaultValue }
```
Comment (sunshinejr) [inline Sources/Mapper.swift]: It won't build now for anyone that want to use this library with Swift 4.0.X, right?

Comment (keith) [reply Sources/Mapper.swift]: Yep, I'll wait for the final Xcode 9.3 release and once I merge this I will bump the major version.

Comment (sunshinejr) [reply Sources/Mapper.swift]: Yeah, I guess that's reasonable 👍

**Comment by Vkt0r** [comment]
@keith Any chance this can be merged soon ?. Xcode 9.3 was released and this is failing on Carthage build.

**Comment by keith** [comment]
@Vkt0r 8.0.0 is tagged with this change.

**Comment by Vkt0r** [comment]
@keith 👍 Thanks for quick response.

**Pull request merged**